providers/rethinkdb.js

```diff
@@ -1,58 +1,55 @@
-// Copyright (c) 2017-2018 dirigeants. All rights reserved. MIT license.
+// Copyright (c) 2017-2019 dirigeants. All rights reserved. MIT license.
 const { Provider, util: { mergeDefault, chunk } } = require('klasa');
-const rethink = require('rethinkdbdash');
+const { r } = require('rethinkdb-ts'); // eslint-disable-line id-length
 
 module.exports = class extends Provider {
 
 	constructor(...args) {
 		super(...args);
-		this.db = rethink(mergeDefault({
-			db: 'test',
-			user: 'admin',
-			password: '',
-			silent: false
-		}, this.client.options.providers.rethinkdb));
+
+		this.db = r;
 	}
 
 	async init() {
-		const { db } = this.db._poolMaster._options;
-		await this.db.branch(this.db.dbList().contains(db), null, this.db.dbCreate(db)).run();
-	}
+		const options = mergeDefault({
+			db: 'test',
+			silent: false
+		}, this.client.options.providers.rethinkdb);
 
-	get exec() {
-		return this.db;
+		this.pool = await r.connectPool(options);
```

Comment:
this.pool is not initialized at the time of calling new ThisClass(...args)

Reply:
(This means adding `this.pool = null;` in the constructor)

Reply:
yeah, I get what ao means :p. fixing now